runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/PCollectionsTest.java

```diff
@@ -154,7 +155,15 @@ public Instant maxTimestamp() {
 
     @Override
     public boolean isCompatible(WindowFn<?, ?> other) {
-      return other != null && this.getClass().equals(other.getClass());
+      return other instanceof CustomWindows;
+    }
+
+    @Override
+    public void verifyCompatibility(WindowFn<?, ?> other) throws IncompatibleWindowException {
+      if (!this.isCompatible(other)) {
+        throw new IncompatibleWindowException(
+            other, "CustomWindows is only compatible with CustomWindows.");
```

Comment:
To link up with IDE support, etc, use `String.format("%s is only compatible with %s", CustomWindows.class.getSimpleName(), CustomWindows.class.getSimpleName())`